bin/lint_commit_msg.py

```diff
@@ -104,6 +105,19 @@ def lint_commit_msg(commit_msg_file, editor):
         print ''
         print '\n'.join(errors)
         print ''
+        if not editor:
+            print (
+                '\n'.join(textwrap.wrap(
+                    'You do not have EDITOR set in your environment. Please '
+                    'set EDITOR in your environment if you want to be able '
+                    'to edit the commit message. In the meantime, you can do:'
+                )) +
+                '\n\n'
+                'git commit --amend\n\n'
+                'to fix issues.'
+            )
+            return 0
```

Comment:
One of the things I fixed in the previous pass that kept biting me was the problem where the commit message effectively disappears if the linter rejects it forcing me to retype the whole thing from scratch. Technically it's in the file `.git/COMMIT-MSG`, but once you start a new commit, that file disappears. So you have to copy the contents of that file before trying the commit again. It's pretty clunky and I don't like re-typing long commit messages.

I agree with what you're thinking here, but I want to make it easy to fix the commit message if there's no `EDITOR` set without having to retype it from scratch. If we simulate an "ignore" here and let the commit message through, then the instructions for fixing the commit message are much easier to walk through.

I'm game for changing the logic here and having a choice with a "no" that causes the linter to reject the message if there's an easy way to allow the user to edit the message rather than having to do it from scratch. Do you know if that's possible?